rbi/prism_static.rbi

```diff
@@ -2,79 +2,79 @@
 
 module Prism
   class ParseResult
```

Comment:
@vinistock do you want to un-nest this? i.e., make it `class `Prism::ParseResult`?

Reply:
Yes, good point. It'll make it easier to spot future issues.

I also removed the `# typed: true` comments from the top of the files, because Tapioca adds those automatically - and we were ending up with duplicates.